### PR TITLE
Fix integration of new icons for types with saved places and refactor to use place ids

### LINKED ISSUE #83: Refactor code to only use place ids for identifying elements
I was only initially providing the name of a place so some of the features use placeName to identify elements whereas other features later on starting using place ids once we started including them. I think it might be nice to change it so that we use place ids to be consistent and to avoid running into issues in the case that two different places somehow have the same name.

### PR DESCRIPTION
**Pull request by PriyaVarra**
This PR fixes some issues with integration of the new icons with the saved places feature.

1. Adds types as data attribute to location cards so that when the user clicks to save the place, it can be parsed and stored in firebase

2. Additionally storing types in firebase for a card

3. Fixing some issues with using the old pins in place of the new pins.

Additionally, this PR replaces all element ids as some variant of an place's place_id (i.e {place_id}-card, {place_id}-icon) to avoid issues with two different places having the same name.

### CODE REVIEW

**Comment by andy7809** [comment]
This is a good catch, thanks Priya!

File: where-to-travel/src/main/webapp/script.js [hunk 1]
```diff
@@ -475,11 +473,11 @@ function addUserDash() {
  * Toggle the focused pin/card off
  */
 function toggleFocusOff() {
-  if (focusedCard != null) {
```
Comment (andy7809) [inline where-to-travel/src/main/webapp/script.js:478]: Good catch here. Thanks!

**Review by andy7809 — COMMENTED** [review]
LGTM. Thanks Priya!

**Comment by saileshsimhadri** [comment]
LGTM

**Pull request merged**